base/MAPL_LocStreamMod.F90

```diff
@@ -404,107 +405,75 @@ subroutine MAPL_LocStreamCreateFromFile(LocStream, LAYOUT, FILENAME, NAME, MASK,
 
 ! Use some heuristics to determine filetype (choices are BINARY and ASCII)
 !------------------------------------------------------------------------
-    ! just get the unit
-    UNIT = GETFILE(FILENAME, DO_OPEN=0, ALL_PES=.true., RC=STATUS)
-    _VERIFY(STATUS)
+    call MAPL_NCIOGetFileType(FILENAME, filetype, _RC)
 
-    INQUIRE(IOLENGTH=IREC) BYTE
-    open (UNIT=UNIT, FILE=FILENAME, FORM='unformatted', ACCESS='DIRECT', RECL=IREC, IOSTAT=status)
-    _VERIFY(STATUS)
-    read (UNIT, REC=1, ERR=100) BYTE
-    call FREE_FILE(UNIT)
-
-    isascii = .true.
-    do i=1,size(byte)
-       if (BYTE(I) < 7) then
-          isascii = .false.
-          exit
-       end if
-    end do
+    isnc4   = (filetype == 0)
+    isascii = (filetype == 1)
+    isbinary= (filetype == 2)
```

Comment:
I wonder if we should make another internal constant for this? 0,1,2 sort of screams for a `MAPL_FILETYPE_NC4` or something.

Then:
```fortran
if(.not. isbinary)
```
could become:
```fortran
if(filetype /= MAPL_FILETYPE_BINARY)
```
?

Reply:
addressed